pkgs/applications/editors/emacs/generic.nix

```diff
@@ -178,6 +179,8 @@ let emacs = (if withMacport then llvmPackages_6.stdenv else stdenv).mkDerivation
       then [ "--disable-ns-self-contained" ]
     else if withX
       then [ "--with-x-toolkit=${toolkit}" "--with-xft" "--with-cairo" ]
+    else if withPgtk
+      then [ "--with-pgtk" "--with-cairo" ]
```

Comment:
This is a sensible change I think but is it important for image support?

Reply:
Fixed.  `--with-cairo` is implied.

Reply:
Just to be clear the rest of the change is to avoid the final `else` that's meant for pure terminal Emacs and will add `--with-png=no --with-jpeg=no` and so on.